src/controllers/logs_test.go

```diff
@@ -0,0 +1,202 @@
+package controllers
+
+import (
+	"context"
+	"github.com/dana-team/platform-backend/src/utils/testutils"
+	"github.com/dana-team/platform-backend/src/utils/testutils/mocks"
+	"go.uber.org/zap"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"strings"
+	"testing"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+func TestFetchCappPodName(t *testing.T) {
+	type args struct {
+		podName string
+		pods    *corev1.PodList
+	}
+	type want struct {
+		name  string
+		found bool
+	}
+
+	// Mock PodList for testing
+	mockPodList := &corev1.PodList{
+		Items: []corev1.Pod{
+			{ObjectMeta: metav1.ObjectMeta{Name: "pod1"}},
+			{ObjectMeta: metav1.ObjectMeta{Name: "pod2"}},
+			{ObjectMeta: metav1.ObjectMeta{Name: "pod3"}},
+		},
+	}
+
+	cases := map[string]struct {
+		args args
+		want want
+	}{
+		"Empty pod name returns first pod": {
+			args: args{
+				podName: "",
+				pods:    mockPodList,
+			},
+			want: want{
+				name:  "pod1",
+				found: true,
+			},
+		},
+		"Existing pod name is found": {
+			args: args{
+				podName: "pod2",
+				pods:    mockPodList,
+			},
+			want: want{
+				name:  "pod2",
+				found: true,
+			},
+		},
+		"Non-existing pod name returns false": {
+			args: args{
+				podName: "nonexistent",
+				pods:    mockPodList,
+			},
+			want: want{
+				name:  "nonexistent",
+				found: false,
+			},
+		},
+		"Empty pod list returns false": {
+			args: args{
+				podName: "pod1",
+				pods:    &corev1.PodList{},
+			},
+			want: want{
+				name:  "pod1",
+				found: false,
+			},
+		},
+	}
+
+	for name, tc := range cases {
+		t.Run(name, func(t *testing.T) {
+			actualName, actualFound := FetchCappPodName(tc.args.podName, tc.args.pods)
+			if actualName != tc.want.name {
+				t.Errorf("Expected pod name %s, but got %s", tc.want.name, actualName)
+			}
+			if actualFound != tc.want.found {
+				t.Errorf("Expected found %v, but got %v", tc.want.found, actualFound)
+			}
+		})
+	}
+}
+
+func TestFetchPodLogs(t *testing.T) {
+	type args struct {
+		client        kubernetes.Interface
+		namespace     string
+		podName       string
+		containerName string
+	}
+	type want struct {
+		errContains string
+	}
+
+	cases := map[string]struct {
+		args args
+		want want
+	}{
+		"ShouldFailGettingLogsOnNonExistingPod": {
+			args: args{
+				client:        fakeClient,
+				namespace:     "test-namespace",
+				podName:       "non-existent-pod",
```

Comment:
Use the constant from the `testutils`

https://github.com/dana-team/platform-backend/blob/9a9f085b8187a05189161f0e38c4a0de621cf2af/src/utils/testutils/consts.go#L28